bundles/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/internal/handler/MieleBridgeHandler.java

```diff
@@ -442,25 +425,19 @@ private List<HomeDevice> getHomeDevices() throws MieleRpcException {
         }
     };
 
-    private @Nullable NetworkInterface getMulticastInterface(String interfaceIpAddress) throws SocketException {
-        Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+    private @Nullable NetworkInterface getMulticastNetworkInterface(String gatewayIpAddress) {
+        try {
+            InetAddress gatewayAddress = InetAddress.getByName(gatewayIpAddress);
 
-        @Nullable
-        NetworkInterface networkInterface;
-        while (networkInterfaces.hasMoreElements()) {
-            networkInterface = networkInterfaces.nextElement();
-            if (networkInterface.isLoopback()) {
-                continue;
-            }
-            for (InterfaceAddress interfaceAddress : networkInterface.getInterfaceAddresses()) {
-                if (logger.isTraceEnabled()) {
-                    logger.trace("Found interface address {} -> {}", interfaceAddress.toString(),
-                            interfaceAddress.getAddress().toString());
-                }
-                if (interfaceAddress.getAddress().toString().endsWith("/" + interfaceIpAddress)) {
-                    return networkInterface;
-                }
+            try (DatagramSocket socket = new DatagramSocket()) {
+                socket.connect(gatewayAddress, 80);
```

Comment:
It seems like you still open a connection to resolve the interface address..?

Reply:
Ah, the "hack" is indeed a very neat trick, then. I tried yesterday to find a way to only trigger the OS interface resolving procedure; it then seems that this *is* what I failed to find.

I know that UDP is connectionless, I just "felt" that `connect()` would lead to some kind of actual network action. But, it makes sense, they could probably have found a better method name than `connect()` though.